plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -414,6 +420,45 @@ public RemoteIterator<LocatedFileStatus> listLocatedStatus(Path path)
         return new S3ObjectsV2RemoteIterator(listPath(path, OptionalInt.empty(), ListingMode.SHALLOW_ALL));
     }
 
+    @Override
+    public void batchDeleteFiles(Collection<Path> paths)
+            throws IOException
+    {
+        try {
+            List<Path> pathsList = paths.stream().collect(toImmutableList());
+            int maxObjectsPerRequest = 1000;
+            List<Path> currentBatch = new ArrayList<>();
+            int currentIndex = 0;
+            for (int i = 0; i < pathsList.size(); i++) {
+                if (currentIndex < maxObjectsPerRequest) {
+                    currentBatch.add(pathsList.get(i));
+                    currentIndex++;
+                }
+                else {
+                    deletePaths(currentBatch);
+                    currentIndex = 0;
+                    currentBatch = new ArrayList<>();
+                }
+            }
+            deletePaths(currentBatch);
+        }
+        catch (AmazonClientException e) {
+            throw new IOException("Exception while batch deleting paths from s3", e);
+        }
+    }
+
+    private void deletePaths(List<Path> paths)
+    {
+        if (!paths.isEmpty()) {
+            List<KeyVersion> keys = paths.stream().map(TrinoS3FileSystem::keyFromPath).map(KeyVersion::new).collect(toImmutableList());
+            DeleteObjectsRequest deleteObjectsRequest = new DeleteObjectsRequest(getBucketName(uri))
+                    .withKeys(keys)
+                    .withQuiet(false);
```

Comment:
what does "not quiet" mean?

Reply:
https://docs.aws.amazon.com/sdkfornet1/latest/apidocs/html/M_Amazon_S3_Model_DeleteObjectsRequest_WithQuiet.htm

But the default is false so I will remove it

Reply:
https://docs.aws.amazon.com/sdkfornet1/latest/apidocs/html/M_Amazon_S3_Model_DeleteObjectsRequest_WithQuiet.htm
I will change it to quiet